Libs/MRML/Core/vtkMRMLCrosshairNode.cxx

```diff
@@ -52,161 +40,28 @@ void vtkMRMLCrosshairNode::WriteXML(ostream& of, int nIndent)
 {
   Superclass::WriteXML(of, nIndent);
 
-  if ( this->CrosshairMode == vtkMRMLCrosshairNode::NoCrosshair )
-    {
-    of << " crosshairMode=\"" << "NoCrosshair" << "\"";
-    }
-  else if ( this->CrosshairMode == vtkMRMLCrosshairNode::ShowBasic )
-    {
-    of << " crosshairMode=\"" << "ShowBasic" << "\"";
-    }
-  else if ( this->CrosshairMode == vtkMRMLCrosshairNode::ShowIntersection )
-    {
-    of << " crosshairMode=\"" << "ShowIntersection" << "\"";
-    }
-  else if ( this->CrosshairMode == vtkMRMLCrosshairNode::ShowHashmarks )
-    {
-    of << " crosshairMode=\"" << "ShowHashmarks" << "\"";
-    }
-  else if ( this->CrosshairMode == vtkMRMLCrosshairNode::ShowAll )
-    {
-    of << " crosshairMode=\"" << "ShowAll" << "\"";
-    }
-  else if ( this->CrosshairMode == vtkMRMLCrosshairNode::ShowSmallBasic )
-    {
-    of << " crosshairMode=\"" << "ShowSmallBasic" << "\"";
-    }
-  else if ( this->CrosshairMode == vtkMRMLCrosshairNode::ShowSmallIntersection )
-    {
-    of << " crosshairMode=\"" << "ShowSmallIntersection" << "\"";
-    }
-
-  if ( this->CrosshairBehavior == vtkMRMLCrosshairNode::OffsetJumpSlice )
-    {
-    of << " crosshairBehavior=\"" << "OffsetJumpSlice" << "\"";
-    }
-  else if (this->CrosshairBehavior == vtkMRMLCrosshairNode::CenteredJumpSlice)
-    {
-    of << " crosshairBehavior=\"" << "CenteredJumpSlice" << "\"";
-    }
-  else if (this->CrosshairBehavior == vtkMRMLCrosshairNode::NoAction)
-    {
-    of << " crosshairBehavior=\"" << "NoAction" << "\"";
-    }
-
-  if ( this->CrosshairThickness == vtkMRMLCrosshairNode::Fine )
-    {
-    of << " crosshairThickness=\"" << "Fine" << "\"";
-    }
-  else if ( this->CrosshairThickness == vtkMRMLCrosshairNode::Medium )
-    {
-    of << " crosshairThickness=\"" << "Medium" << "\"";
-    }
-  else if ( this->CrosshairThickness == vtkMRMLCrosshairNode::Thick )
-    {
-    of << " crosshairThickness=\"" << "Thick" << "\"";
-    }
-
-  of <<  " crosshairRAS=\"" << this->CrosshairRAS[0] << " "
-     << this->CrosshairRAS[1] << " " << this->CrosshairRAS[2] << "\"";
+  vtkMRMLWriteXMLBeginMacro(of);
+  vtkMRMLWriteXMLEnumMacro(crosshairMode, CrosshairMode);
+  vtkMRMLWriteXMLEnumMacro(crosshairBehavior, CrosshairBehavior);
+  vtkMRMLWriteXMLEnumMacro(crosshairThickness, CrosshairThickness);
+  vtkMRMLWriteXMLVectorMacro(crosshairRAS, CrosshairRAS, double, 3);
+  vtkMRMLWriteXMLBooleanMacro(fastPick3D, FastPick3D);
```

Comment:
```suggestion
  // This property is only for evaluation of this feature and the value is not stored persistently in the scene file.
  // vtkMRMLWriteXMLBooleanMacro(fastPick3D, FastPick3D);
```

Reply:
Makes sense, I'll change this.

Reply:
Looks like the `fastPick3D` property is still serialized ?

Reply:
Sorry, it seems that some of the changes have been discarded during rebase or force-push.